docs/authentication.md

```diff
@@ -0,0 +1,128 @@
+# A Guide on ScalarDL Authentication
+
+This document explains the ScalarDL authentication mechanism and how to use it properly. 
+
+## Authentication in ScalarDL
+
+Authentication is one of the key roles in ScalarDL to make the protocol work as expected.
+ScalarDL uses authentication for the following three places:
+* Client authentication (for Ledger and Auditor)
+    * Client authentication authenticates clients by Ledger and Auditor using client-generated signatures attached to requests from clients.
+* Ledger authentication (for Auditor)
+    * Ledger authentication authenticates Ledger by Auditor using Ledger-generated signatures attached to [Asset Proofs](how-to-use-proof.md).
+* Auditor authentication (for Ledger )
```

Comment:
I am not sure if this is a good revision but this makes me easier to understand.
- Client authentication (by Ledger and Auditor)
 - Ledger and Auditor use the client-generate signatures attached to request to authenticate clients
- Auditor authentication (by Ledger)
 - Ledger uses Auditor-generated signatures attached to client requests to authenticate Auditor
- Ledger authentication (by Auditor)
 - Auditor uses Ledger-generated signatures attached to Asset Proof to authenticate Ledger

Reply:
Agreed. Thank you!